spec/controllers/picture_controller_spec.rb

```diff
@@ -0,0 +1,34 @@
+require "spec_helper"
+
+describe PictureController do
+  let(:picture_content) { "BINARY IMAGE CONTENT" }
+  let(:service_picture) { FactoryGirl.create(:picture, :extension => "jpg") }
+
+  before(:each) do
+    set_user_privileges
+
+    FactoryGirl.create(:vmdb_database)
+    EvmSpecHelper.create_guid_miq_server_zone
+
+    service_picture.content = picture_content
+    service_picture.save
+  end
+
+  it 'can service a picture directly from the database' do
+    visit "/pictures/#{service_picture.compressed_id}.#{service_picture.extension}"
+    expect(page.status_code).to eq(200)
+    expect(response.body).to eq(picture_content)
+  end
+
+  it "responds with a Not Found with pictures of incorrect extension" do
+    visit "/pictures/#{service_picture.compressed_id}.png"
+    expect(page.status_code).to eq(404)
+    expect(response.body).to be_blank
+  end
+
+  it "responds with a Not Found with unknown pictures" do
+    visit "/pictures/bogusimage.gif"
+    expect(page.status_code).to eq(404)
+    expect(response.body).to be_blank
+  end
+end
```

Comment:
Are the specs using memcache?  if so, you should clear memcache after every test to ensure you don't reuse it.